test/unit/sessions.test.js

```diff
@@ -256,20 +256,37 @@ describe('Sessions - unit', function () {
       describe('from an implicit session', () => {
         const session = new ClientSession(topology, serverSessionPool, { explicit: false }); // make an implicit session
 
-        it('should throw if the session hasEnded before serverSession was acquired', () => {
+        it('should throw if the session ended before serverSession was acquired', () => {
           expect(session).to.have.property(serverSessionSymbol, null);
           session.hasEnded = true;
           expect(() => session.serverSession).to.throw(MongoRuntimeError);
         });
 
-        it('should acquire a serverSession if clientSession.hadEnded is false', () => {
+        it('should acquire a serverSession if clientSession.hasEnded is false and serverSession is not set', () => {
           expect(session).to.have.property(serverSessionSymbol, null);
           session.hasEnded = false;
           const acquireSpy = sinon.spy(serverSessionPool, 'acquire');
           expect(session.serverSession).to.be.instanceOf(ServerSession);
           expect(acquireSpy.calledOnce).to.be.true;
           acquireSpy.restore();
         });
+
+        it('should return the existing serverSession and not acquire a new one if one is already set', () => {
```

Comment:
can you make this test the case where the session has ended, too?

Reply:
done!